pkg/controller/nodeipam/ipam/cloud_cidr_allocator.go

```diff
@@ -277,13 +277,53 @@ func (ca *cloudCIDRAllocator) updateCIDRAllocation(nodeName string) error {
 	}
 
 	cidrStrings := make([]string, 0)
-	var northInterfaces networkv1.NorthInterfacesAnnotation
-	var additionalNodeNetworks networkv1.MultiNetworkAnnotation
 
 	if len(instance.NetworkInterfaces) == 0 || (len(instance.NetworkInterfaces) == 1 && len(instance.NetworkInterfaces[0].AliasIpRanges) == 0) {
 		nodeutil.RecordNodeStatusChange(ca.recorder, node, "CIDRNotAvailable")
 		return fmt.Errorf("failed to allocate cidr: Node %v has no ranges from which CIDRs can be allocated", node.Name)
 	}
+
+	defer func() {
+		if rerr != nil {
+			return
+		}
+		// update node's spec
+		if !reflect.DeepEqual(node.Spec, oldNode.Spec) {
+			for i := 0; i < cidrUpdateRetries; i++ {
+				if err = utilnode.PatchNodeCIDRs(ca.client, types.NodeName(node.Name), node.Spec.PodCIDRs); err == nil {
+					klog.InfoS("Set the node PodCIDRs", "nodeName", node.Name, "cidrStrings", node.Spec.PodCIDRs)
+					break
+				}
+			}
+			if err != nil {
+				nodeutil.RecordNodeStatusChange(ca.recorder, node, "CIDRAssignmentFailed")
+				klog.ErrorS(err, "Failed to update the node PodCIDR after multiple attempts", "nodeName", node.Name, "cidrStrings", node.Spec.PodCIDRs)
+				rerr = multierror.Append(rerr, err)
+				return
+			}
+			err = utilnode.SetNodeCondition(ca.client, types.NodeName(node.Name), v1.NodeCondition{
+				Type:               v1.NodeNetworkUnavailable,
+				Status:             v1.ConditionFalse,
+				Reason:             "RouteCreated",
+				Message:            "NodeController create implicit route",
+				LastTransitionTime: metav1.Now(),
+			})
+			if err != nil {
+				klog.ErrorS(err, "Error setting route status for the node", "nodeName", node.Name)
+			}
+		}
+
+		// update node's annotations and capacity
+		if !reflect.DeepEqual(node.Annotations, oldNode.Annotations) {
+			if err = utilnode.PatchNodeMultiNetwork(ca.client, node); err != nil {
+				nodeutil.RecordNodeStatusChange(ca.recorder, node, "CIDRAssignmentFailed")
```

Comment:
Do you know the difference between CIDRNotAvailable and CIDRAssignmentFailed. The earlier code was setting CIDRNotAvailable for this error. I don't know the implications of this change.

Reply:
I do not know the diff between the 2 events. The previous code was setting "CIDRAssignmentFailed", see line 444 of the old file.